From 291d22030533a529bf8fe96ac0290d672d3b21c8 Mon Sep 17 00:00:00 2001
From: Eric Fourmont <eric.fourmont-ext@st.com>
Date: Mon, 22 Apr 2024 17:48:22 +0200
Subject: [PATCH] kbuild: stop build if (external) device tree compilation
 fails

Stop the build with error as soon as there is an error with device tree
compilation. Either with an "internal" device tree with or without valid
KBUILD_EXTDTS, or with an external device tree with valid KBUILD_EXTDTS.

Change-Id: Ib6b8aed12619b1c00d16a2c0c586b2d165dc76e6
Signed-off-by: Christophe Priouzeau <christophe.priouzeau@foss.st.com>
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/368451
Reviewed-by: Christophe PRIOUZEAU <christophe.priouzeau@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 4562dacf5c8c..5f95d03de6fa 100644
--- a/Makefile
+++ b/Makefile
@@ -1410,7 +1410,7 @@ KBUILD_EXTDTB := $(or $(and $(wildcard $(KBUILD_EXTDTS)), 1), 0)
 	test ! -e $(dtstree)/$@ -a $(KBUILD_EXTDTB) -eq 1  && ( \
 		echo "Looking for $(patsubst st/%.dtb,%.dtb,$@) into $(KBUILD_EXTDTS)"; \
 		$(MAKE) $(build)=$(stdtstree) src=$(KBUILD_EXTDTS) $(dtstree)/$@ || \
-		/bin/true) || /bin/true)
+		/bin/false) || /bin/false)
 
 %.dtbo: dtbs_prepare
 	$(Q)$(MAKE) $(build)=$(dtstree) $(dtstree)/$@
@@ -1418,10 +1418,10 @@ KBUILD_EXTDTB := $(or $(and $(wildcard $(KBUILD_EXTDTS)), 1), 0)
 PHONY += dtbs dtbs_prepare dtbs_install dtbs_check
 dtbs: dtbs_prepare
 	$(Q)$(MAKE) $(build)=$(stdtstree)
-	$(Q)test $(KBUILD_EXTDTB) -eq 1 && ( \
+	$(Q)test $(KBUILD_EXTDTB) -eq 0 || ( \
 		echo "Looking for device trees into $(KBUILD_EXTDTS)"; \
 		$(MAKE) $(build)=$(stdtstree) src=$(KBUILD_EXTDTS) || \
-		/bin/true) || /bin/true
+		/bin/false)
 
 # include/config/kernel.release is actually needed when installing DTBs because
 # INSTALL_DTBS_PATH contains $(KERNELRELEASE). However, we do not want to make
-- 
2.39.5

